lib/src/main/kotlin/com/swmansion/starknet/data/serializers/GatewayTransactionTransformingSerializer.kt

```diff
@@ -0,0 +1,15 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.transactions.Transaction
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonTransformingSerializer
+import kotlinx.serialization.json.jsonObject
+
+object GatewayTransactionTransformingSerializer :
+    JsonTransformingSerializer<Transaction>(TransactionPolymorphicSerializer) {
```

Comment:
Don't know if there's a better way of doing this, but turns out you can just pass any serializer you won't to be used after transformation as argument. Don't know why I didn't think of that before.

Reply:
🪄